pkg/crd/markers/crd.go

```diff
@@ -345,3 +349,29 @@ func (s DeprecatedVersion) ApplyToCRD(crd *apiext.CustomResourceDefinitionSpec,
 	}
 	return nil
 }
+
+// +controllertools:marker:generateHelp:category=CRD
+
+// APIApprovedKubernetesGroup configures the api-approved.kubernetes.io for a CRD
+// with Kubernetes groups, including "k8s.io", "*.k8s.io", "kubernetes.io", "*.kubernetes.io".
+type APIApprovedKubernetesGroup struct {
+	// URL is the value of api-approved.kubernetes.io annotation on CRD,
+	// it should be a valid URL which contains Scheme and Host and
+	// should not start with "unapproved".
+	URL string `marker:"URL"`
+}
```

Comment:
Any chance we could make this a more generic `Annotation` struct with `Key` and `Value` fields that we parse out of the marker?

I can imagine a few other use cases, for example adding annotation `cert-manager.io/inject-ca-from-secret` to the CRD when it needs a CA injected for a conversion webhook configuration.

Reply:
Thanks, it makes sense. I will update this later.